src/core/Akka/Dispatch/Mailboxes.cs

```diff
@@ -76,23 +93,53 @@ public Type GetMailboxType(Props props, Config dispatcherConfig)
                 return FromConfig(props.Mailbox);
             }
 
-            var actortype = props.Type;
-            var interfaces = actortype.GetInterfaces()
-                .Where(i => i.IsGenericType)
-                .Where(i => i.GetGenericTypeDefinition() == typeof (IRequiresMessageQueue<>))
-                .Select(i => i.GetGenericArguments().First())
-                .ToList();
+            if (dispatcherConfig == null)
+                dispatcherConfig = ConfigurationFactory.Empty;
+            var id = dispatcherConfig.GetString("id");
+            var deploy = props.Deploy;
+            var actorType = props.Type;
+            var actorRequirement = new Lazy<Type>(() => GetRequiredType(actorType));
+
+            var mailboxRequirement = GetMailboxRequirement(dispatcherConfig);
+            var hasMailboxRequirement = mailboxRequirement != typeof (IMessageQueue);
+
+            var hasMailboxType = dispatcherConfig.HasPath("mailbox-type") &&
+                                 dispatcherConfig.GetString("mailbox-type") != Deploy.NoMailboxGiven;
 
-            if (interfaces.Count > 0)
+            Func<Type, Type> verifyRequirements = mailboxType =>
             {
-                var config = _requirementsMapping[interfaces.First()];
-                var mailbox = config.GetString("mailbox-type");
-                var mailboxType = Type.GetType(mailbox);
+                // TODO Akka code after introducing IProducesMessageQueue
```

Comment:
Do we want to enforce the requirements anytime soon? If so another issue should be created to complete this.